src/vm/ast/definition_sorter/mod.rs

```diff
@@ -213,11 +213,12 @@ impl<'a> DefinitionSorter {
                             }
                         } else if let Some(native_function) =
                             // NOTE: can use ClarityVersion::latest() here only as long as NO NEW FUNCTIONS are special cased
-                        //        in the definition sorter.
+                            //        in the definition sorter.
                             NativeFunctions::lookup_by_name_at_version(
-                            function_name,
-                            &ClarityVersion::latest(),
-                        ) {
+                                    function_name,
+                                    &ClarityVersion::latest(),
```

Comment:
Hold up -- you need to gate this.  The Clarity version you should be passing here is whatever the transaction's `GlobalContext` is running in.

Reply:
I think this came up in a different PR review -- the definition sorter is not a "normal" analysis pass (it's not an analysis pass at all -- that's why it's part of the AST module). All it is trying to do in this loop is figure out whether or not a particular function application is a "special" application where the arguments aren't dereferenced (e.g., the tuple keys in a tuple constructor). As long as the new Clarity version doesn't add a new branch to this match, it is fine to just use `ClarityVersion` latest here. However, it may make sense to future-proof this better by gating the `definition-sorter`, but I think that's sufficiently different that it should be done in it's own PR. Issue here: https://github.com/blockstack/stacks-blockchain/issues/2759